src/stateMachines/services/oid4vciMachineService.ts

```diff
@@ -0,0 +1,191 @@
+import {URL} from 'react-native-url-polyfill';
+import {v4 as uuidv4} from 'uuid';
+import {CompactJWT, VerifiableCredential} from '@veramo/core';
+import {computeEntryHash} from '@veramo/utils';
+import {CredentialResponse, CredentialSupported} from '@sphereon/oid4vci-common';
+import {CorrelationIdentifierEnum, IBasicCredentialLocaleBranding, IBasicIdentity, IContact, IdentityRoleEnum} from '@sphereon/ssi-sdk.data-store';
+import {
+  CredentialMapper,
+  IIssuer,
+  IVerifiableCredential,
+  OriginalVerifiableCredential,
+  W3CVerifiableCredential,
+  WrappedVerifiableCredential,
+} from '@sphereon/ssi-types';
+import OpenId4VcIssuanceProvider, {CredentialFromOffer} from '../../providers/credential/OpenId4VcIssuanceProvider';
+import {addCredentialBranding, selectAppLocaleBranding} from '../../services/brandingService';
+import {verifyCredential} from '../../services/credentialService';
+import {getContacts} from '../../services/contactService';
+import store from '../../store';
+import {storeVerifiableCredential} from '../../store/actions/credential.actions';
+import {addIdentity} from '../../store/actions/contact.actions';
+import {ICredentialTypeSelection, IVerificationResult, MappedCredentialOffer, OID4VCIMachineContext, OID4VCIMachineEventTypes} from '../../types';
+
+export const initiating = async (_ctx: OID4VCIMachineContext, _event: OID4VCIMachineEventTypes): Promise<OpenId4VcIssuanceProvider> => {
+  const {requestData} = _ctx;
+  return OpenId4VcIssuanceProvider.initiationFromUri({uri: requestData.uri});
+};
+
+export const createCredentialSelection = async (
+  _ctx: OID4VCIMachineContext,
```

Comment:
Is there a need to provide the entire context as argument?

Reply:
This and the next comment about the event applies to all functions in the file

Reply:
well, the onboarding service is doing the same. and i am still on the fence if these are the same services as any other service we have. So personally i am fine with a tighter coupling to the machines. i do not see these services being used anywhere else. or we should start splitting things up even more. because for example if i want to have some supported credentials wnywehere in the app. i would not expect calling a oid4vci machine service.

Reply:
also isnt this how a service from a machine is called?

Reply:
Touché. I guess in the future we can limit the context to only include the keys/properties of the context that are really needed by the method. In other words use Pick<Context, 'prop'>. Then the service itself only knows about the specific properties, but the machine would pass in the entire context

Reply:
Changed the types to use Pick